library/VTKExtensions/Rendering/vtkF3DInteractorStyle.cxx

```diff
@@ -114,7 +114,19 @@ void vtkF3DInteractorStyle::Dolly()
   {
     return;
   }
-  this->Superclass::Dolly();
+
+  vtkRenderWindowInteractor* rwi = this->Interactor;
+  const double* center = this->CurrentRenderer->GetCenter();
+  const int dy = rwi->GetEventPosition()[1] - rwi->GetLastEventPosition()[1];
+  const int dx = rwi->GetEventPosition()[0] - rwi->GetLastEventPosition()[0];
+  auto dt = std::abs(dy) > std::abs(dx) ? dy : dx;
+  vtkF3DRenderer* ren = vtkF3DRenderer::SafeDownCast(this->CurrentRenderer);
+  if (ren && ren->GetInvertZoom())
+  {
+    dt *= -1;
+  }
+  const double dtf = this->MotionFactor * dt / center[1];
```

Comment:
Shouldn't `center[0]` come into play now that we're looking at both axes? Maybe picking the max later
```cpp
const double dxf = this->MotionFactor * dx / center[0];
const double dyf = this->MotionFactor * dy / center[1];
double dtf = std::abs(dyf) > std::abs(dxf) ? dyf : dxf;
if (ren && ren->GetInvertZoom())
{
    dtf *= -1;
}
```
or does that make no difference?

Reply:
@snoyer Wouldn't this only change the magnitude of the dolly move?

Reply:
I think @snoyer is right.